apis/datadoghq/v1alpha1/datadogagent_default.go

```diff
@@ -133,32 +133,46 @@ func DefaultDatadogAgent(dda *DatadogAgent) *DatadogAgentStatus {
 	return dso
 }
 
-func defaultCredentials(ddaSpec *DatadogAgentSpec, dso *DatadogAgentStatus) {
-	if ddaSpec.Credentials == nil {
-		ddaSpec.Credentials = &AgentCredentials{}
+func defaultCredentials(dda *DatadogAgent, dso *DatadogAgentStatus) {
+	if dda.Spec.Credentials == nil {
+		dda.Spec.Credentials = &AgentCredentials{}
 	}
-	if ddaSpec.Credentials.UseSecretBackend == nil {
-		ddaSpec.Credentials.UseSecretBackend = apiutils.NewBoolPointer(false)
+
+	if dda.Spec.Credentials.UseSecretBackend == nil {
+		dda.Spec.Credentials.UseSecretBackend = apiutils.NewBoolPointer(false)
 		if dso.DefaultOverride == nil {
 			dso.DefaultOverride = &DatadogAgentSpec{}
 		}
 		dso.DefaultOverride.Credentials = &AgentCredentials{
-			UseSecretBackend: ddaSpec.Credentials.UseSecretBackend,
+			UseSecretBackend: dda.Spec.Credentials.UseSecretBackend,
 		}
 	}
 
-	// Generate a Token for clusterAgent-Agent communication if not provided
-	if ddaSpec.Credentials.Token == "" {
-		if dso.ClusterAgent == nil {
-			dso.ClusterAgent = &DeploymentStatus{}
-		}
-		if dso.ClusterAgent.GeneratedToken == "" {
-			dso.ClusterAgent.GeneratedToken = apiutils.GenerateRandomString(32)
-		}
-	} else if dso.ClusterAgent != nil {
-		// cleanup the generated token if after an update the user specify one
-		// in the spec.Credentials.Token
-		dso.ClusterAgent.GeneratedToken = ""
+	defaultClusterAgentToken(dda, dso)
+}
+
+func defaultClusterAgentToken(dda *DatadogAgent, dso *DatadogAgentStatus) {
```

Comment:
IMO this change is not fully backward compatible. 
what happen if the `dso.ClusterAgent.GeneratedToken` was already defined? 

can we add another step in the function to get the token present in `status.ClusterAgent.GeneratedToken` if present ?

Reply:
I added a new commit with this :+1: